nmpolicy/internal/resolver/terminal.go

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright 2001 NMPolicy Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at:
+ *
+ *	  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package resolver
+
+import (
+	"fmt"
+
+	"github.com/nmstate/nmpolicy/nmpolicy/internal/ast"
+)
+
+func newTerminalFromInterface(iface interface{}) (*ast.Terminal, error) {
+	switch v := iface.(type) {
+	case string:
+		return &ast.Terminal{String: &v}, nil
+	case int:
+		return &ast.Terminal{Number: &v}, nil
```

Comment:
[filter](https://github.com/nmstate/nmpolicy/blob/bce7aa71e17cf80bc97586f0d7d780652d727263/nmpolicy/internal/resolver/filter.go#L43) should be updated as well since it currently doesn't support `Number`s.

Reply:
For the main case we are using numbers for the path but not for the value filter with, we can do that at a follow up.

Reply:
So it doesn't make sense to put the number resolving here as well since it is used for the filtered value and not the path.